src/linux/net/wifi/core/AccessPointControllerLinux.cxx

```diff
@@ -121,19 +124,35 @@ AccessPointControllerLinux::SetOperationalState(AccessPointOperationalState oper
 
     switch (operationalState) {
     case AccessPointOperationalState::Enabled: {
+        AUDIT_LOGD << std::format("Attempting to set operational state of AP {} to 'enabled'", status.AccessPointId);
         try {
             m_hostapd.Enable();
             status.Code = AccessPointOperationStatusCode::Succeeded;
+
+            // Validate that enabling the operational state succeeds.
+            AccessPointOperationalState actualOperationalState{};
+            auto getOperationalStateStatus = GetOperationalState(actualOperationalState);
+            if (getOperationalStateStatus.Code == AccessPointOperationStatusCode::Succeeded && actualOperationalState == AccessPointOperationalState::Enabled) {
+                AUDIT_LOGI << std::format("Operational state of AP {} set to 'enabled'", status.AccessPointId);
+            }
         } catch (const Wpa::HostapdException& ex) {
             status.Code = AccessPointOperationStatusCode::InternalError;
             status.Details = std::format("failed to set operational state to 'enabled' ({})", ex.what());
         }
         break;
     }
     case AccessPointOperationalState::Disabled: {
+        AUDIT_LOGD << std::format("Attempting to set operational state of AP {} to 'disabled'", status.AccessPointId);
         try {
             m_hostapd.Disable();
             status.Code = AccessPointOperationStatusCode::Succeeded;
+
+            // Validate that disabling the operational state succeeds.
+            AccessPointOperationalState actualOperationalState{};
+            auto getOperationalStateStatus = GetOperationalState(actualOperationalState);
+            if (getOperationalStateStatus.Code == AccessPointOperationStatusCode::Succeeded && actualOperationalState == AccessPointOperationalState::Disabled) {
+                AUDIT_LOGI << std::format("Operational state of AP {} set to 'disabled'", status.AccessPointId);
+            }
```

Comment:
This code won't run if the call to `Disable` throws an exception. The `Hostapd` class signals errors by throwing a `HostapdException` instead of by return value or output parameter, so I think this will need to be accounted for here.

Reply:
That was intentional because I assumed that the state wouldn't change to 'disabled' if `Disable` fails/throws, so I put this in the `try` block, but maybe it's better to move this outside of the `try/catch`